src/main/java/com/softserveinc/dokazovi/handlers/PostDeleteEventHandler.java

```diff
@@ -0,0 +1,21 @@
+package com.softserveinc.dokazovi.handlers;
+
+import com.softserveinc.dokazovi.dto.post.PostTitleDTO;
+import com.softserveinc.dokazovi.events.PostDeleteEvent;
+import org.springframework.context.ApplicationListener;
+import org.springframework.stereotype.Component;
+
+@Component
+public class PostDeleteEventHandler implements ApplicationListener<PostDeleteEvent> {
+
+    private PostTitleDTO postTitleDTO;
+
+    @Override
+    public void onApplicationEvent(PostDeleteEvent event) {
+        this.postTitleDTO = event.getPostTitleDTO();
+    }
+
+    public PostTitleDTO getPostTitleDTO() {
```

Comment:
This is risky - in the current setup, we are running Apache Tomcat.
This means that we have a thread pool, which, as far as I am aware, is equal to number of CPU cores in terms of size.
This means that this will fail during even small load.

You should move logic related to handling PostDeleteEvent into this class, don't be shy to include even calls to services which work with the database.

If you do it well, it will also lower coupling much.

By the way, by marking `onApplicationEvent` with `@Async`, you will schedule it be be run in a separate task of the thread pool, which will decouple it from the event publisher even further.